File: backend/src/main/java/com/gxs/bank/service/ContactService.java
package com.gxs.bank.service;

import com.gxs.bank.dto.request.ContactRequest;
import com.gxs.bank.model.ContactSubmission;
import com.gxs.bank.repository.ContactRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class ContactService {

    private final ContactRepository contactRepository;

    public ContactSubmission submit(ContactRequest request) {
        ContactSubmission submission = ContactSubmission.builder()
                .name(request.getName())
                .email(request.getEmail())
                .subject(request.getSubject())
                .message(request.getMessage())
                .status(ContactSubmission.Status.NEW)
                .build();

        return contactRepository.save(submission);
    }

    public List<ContactSubmission> getByEmail(String email) {
        return contactRepository.findByEmailOrderByCreatedAtDesc(email);
    }

    public List<ContactSubmission> getAll() {
        return contactRepository.findAllByOrderByCreatedAtDesc();
    }
}

File: backend-python/com/gxs/bank/service/ContactService.py
from com.gxs.bank.dto.request.ContactRequest import ContactRequest
from com.gxs.bank.model.ContactSubmission import ContactSubmission
from com.gxs.bank.repository.ContactRepository import ContactRepository


class ContactService:
    def __init__(self, db):
        self.db = db
        self.contactRepository = ContactRepository(db)

    def submit(self, request: ContactRequest):
        submission = ContactSubmission(
            name=request.name,
            email=request.email,
            subject=request.subject,
            message=request.message,
            status=ContactSubmission.Status.NEW,
        )
        self.contactRepository.save(submission)
        self.db.commit()
        self.db.refresh(submission)
        return submission

    def getByEmail(self, email: str):
        return self.contactRepository.findByEmailOrderByCreatedAtDesc(email)

    def getAll(self):
        return self.contactRepository.findAllByOrderByCreatedAtDesc()
